### PR TITLE
fixed bad file include endings and wrong library linking

### PR DESCRIPTION
**Pull request by bloesch**

### CODE REVIEW

**Comment by schneith** [comment]
lgtm. sorry for the bugs:)

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,19 +24,19 @@ cs_add_executable(filter_test_node
 
 #add_subdirectory(src/test)
 
-target_link_libraries(filter_test_node ${PROJECT_NAME})
+target_link_libraries(filter_test_node ${PROJECT_NAME} ${catkin_LIBRARIES})
```
Comment (helenol) [inline CMakeLists.txt]: Shouldn't be necessary (`${catkin_LIBRARIES}`) as cs_add_library does this for you.

File: include/filter_test/defines.h [hunk 2]
```diff
@@ -12,10 +12,8 @@
 
 namespace tsif {
 
-#define TSIF_VERBOSE true
-
-#define DO_SANITY_CHECKS true
-#define VERBOSE_MODE true  // Should we output a lot of stuff to the console?
+#define TSIF_VERBOSE \
```
Comment (helenol) [inline include/filter_test/defines.h]: Why the line break?

File: include/filter_test/helper_functions.h [hunk 3]
```diff
@@ -23,7 +23,8 @@ inline std::string padTo(
 
 // Find a value in the vector. If you use this function often maybe consider
 // switching to a set!
```
Comment (helenol) [inline include/filter_test/helper_functions.h:25]: Nice comment. :)

File: include/filter_test/residuals/constant_velocity_residual.h [hunk 4]
```diff
@@ -166,6 +161,17 @@ class ConstantVelocityResidual : public ResidualBase {
...
 
+  virtual bool checkJacobians(
+      const VectorOfBlocks& state1, const VectorOfBlocks& state2,
+      const int64_t t1_ns, const int64_t t2_ns, const double delta) {
+    const std::vector<const TimedMeasurementVector*>
+        measurement_vectors;  // Create measurements.
+    // Since residual is not depending on any measurements this is trivial :)
+    return checkJacobiansImpl(
+        state1, state2, measurement_vectors, t1_ns, t2_ns, delta);
+    ;
```
Comment (helenol) [inline include/filter_test/residuals/constant_velocity_residual.h]: Unnecessary

File: src/library/residual.cpp [hunk 5]
```diff
@@ -12,4 +12,83 @@ namespace tsif {
...
+        delta_vector[index_to_test] = delta;
+        // boxplus of relevant block
+        block_helper::boxPlus(
+            state1, state1_minimal_dimension, delta_vector, &state1_perturbed);
+        evaluate(
+            state1_perturbed, state2, measurement_vectors, t1_ns, t2_ns,
+            residual_perturbed, nullptr, nullptr);
+
+        (*jacobian_wrt_state1)[block_index].col(block_offset) =
+            (1 / delta) * (residual_perturbed - residual);
```
Comment (helenol) [inline src/library/residual.cpp:108]: Any checks that delta is not-0 or not within tolerance of 0? Is this necessary?

Comment (helenol) [reply src/library/residual.cpp:108]: I guess if you're taking finite difference with 0 or double epsilon delta then you deserve weird results or a seg fault. ;)

Comment (burrimi) [reply src/library/residual.cpp:108]: added check

**Pull request merged**